### PR TITLE
64 set up GitHub lfs for testing, and ruff for linting

### PR DESCRIPTION
**Pull request by DSilva27**

### CODE REVIEW

**Comment by DSilva27** [comment]
This PR sets up git lfs for the testing data. 

Small changes

- Updated workflows to reflect the git lfs addition
- Added a linting test to the workflow pipeline

File: tests/scripts/fetch_test_data.sh
Comment (geoffwoollard) [inline tests/scripts/fetch_test_data.sh:1]: without this file, how does someone get files locally for running local tests?

if they can get it through git with some special command, we should put in some documentation for that.

**Review by geoffwoollard — APPROVED** [review]

File: tests/scripts/fetch_test_data.sh
Comment (DSilva27) [reply tests/scripts/fetch_test_data.sh:1]: They have to install git lfs and then when they clone they get the files. I mentioned this in the README under the installation for developers. I think I will do a rewrite of the README soon anyway, so I will make this more clear when I do that.

Comment (geoffwoollard) [reply tests/scripts/fetch_test_data.sh:1]: ok, let's wait to merge to main for that.

**Pull request merged**